File: frontend/src/services/taskService.ts
import api from './api'
import { Task, CreateTaskRequest, UpdateTaskRequest } from '../types'

export const taskService = {
  async getAllTasks(): Promise<Task[]> {
    const response = await api.get<Task[]>('/tasks')
    return response.data
  },

  async getMyTasks(): Promise<Task[]> {
    const response = await api.get<Task[]>('/tasks/my-tasks')
    return response.data
  },

  async getTaskById(id: number): Promise<Task> {
    const response = await api.get<Task>(`/tasks/${id}`)
    return response.data
  },

  async createTask(request: CreateTaskRequest): Promise<Task> {
    const response = await api.post<Task>('/tasks', request)
    return response.data
  },

  async updateTask(id: number, request: UpdateTaskRequest): Promise<Task> {
    const response = await api.put<Task>(`/tasks/${id}`, request)
    return response.data
  },

  async deleteTask(id: number): Promise<void> {
    await api.delete(`/tasks/${id}`)
  },
}

File: backend/src/main/java/com/personalweb/app/service/TaskService.java
package com.personalweb.app.service;

import com.personalweb.app.dto.CreateTaskRequest;
import com.personalweb.app.dto.TaskDto;
import com.personalweb.app.dto.UpdateTaskRequest;

import java.util.List;

public interface TaskService {

    List<TaskDto> getAllTasks();

    List<TaskDto> getTasksByUser(String username);

    TaskDto getTaskById(Long id);

    TaskDto createTask(CreateTaskRequest request, String createdByUsername);

    TaskDto updateTask(Long id, UpdateTaskRequest request);

    void deleteTask(Long id);
}
